FlowCrypt/Functionality/Mail Provider/Threads/MessagesThreadOperationsProvider.swift

```diff
@@ -71,7 +71,14 @@ extension GmailService: MessagesThreadOperationsProvider {
     }
 
     func archive(thread: MessageThread, in folder: String) async throws {
-        try await update(thread: thread, labelsToRemove: [.inbox, .sent])
+        try await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            for message in thread.messages {
```

Comment:
What's advantage of manually updating each message vs using `update(thread:...`?

I thought `GTLRGmail_ModifyThreadRequest` updates each message too (as documentation says - https://developers.google.com/gmail/api/reference/rest/v1/users.threads/modify)

Reply:
Well in this task, Tom mentioned archive thread feature is too aggressive.
If we just call `GTLRGmail_ModifyThreadRequest`, it would archive all emails in this thread (at the time user taps archive button).
But with updated logic, it would only archive messages which are currently displayed.


> It seems it will also archive the email that was received while user was reading the previous email in the conversation.
> 
> Expected: only archive emails that were present in the conversation the moment user opened the conversation, not the moment they decide to archive
> 
> I believe that will bring it closer to Gmail app behavior. Right after the archive + refresh of inbox, the conversation should still be there, with one unread email.
> 
>

Reply:
Sounds good 👍